api/preprints/views.py

```diff
@@ -0,0 +1,197 @@
+from rest_framework import generics
+from rest_framework import permissions as drf_permissions
+
+from modularodm import Q
+
+from framework.auth.oauth_scopes import CoreScopes
+
+from website.models import Node
+
+from api.base import permissions as base_permissions
+from api.base.views import JSONAPIBaseView
+from api.base.filters import ODMFilterMixin
+from api.preprints.parsers import PreprintsJSONAPIParser, PreprintsJSONAPIParserForRegularJSON
+from api.preprints.serializers import PreprintSerializer, PreprintDetailSerializer, PreprintDetailRetrieveSerializer
+from api.nodes.views import NodeMixin, WaterButlerMixin, NodeContributorsList, NodeContributorsSerializer
+from api.base.utils import get_object_or_error
+from rest_framework.exceptions import NotFound
+
+
+class PreprintMixin(NodeMixin):
+    serializer_class = PreprintSerializer
+    node_lookup_url_kwarg = 'node_id'
+
+    def get_node(self):
+        node = get_object_or_error(
+            Node,
+            self.kwargs[self.node_lookup_url_kwarg],
+            display_name='preprint'
+        )
+        if not node.is_preprint and self.request.method != 'POST':
+            raise NotFound
+
+        return node
+
+
+class PreprintList(JSONAPIBaseView, generics.ListAPIView, ODMFilterMixin):
```

Comment:
As mentioned above, this should perhaps be a `ListCreateAPIView`, rather than having the `Detail` view perform the create. If this is too big of an architectural change at this point, please include something like `"Under no circumstances should anyone expect this route to be stable"` in the docstring (of bot this and `PreprintDetail`).